src/widget/wtrackmenu.cpp

```diff
@@ -0,0 +1,1434 @@
+        pTrack->resetKeys();
+    }
+}
+
+void WTrackMenu::slotClearReplayGain() {
+    const TrackPointerList trackPointers = getTrackPointers();
+    if (trackPointers.empty()) {
+        return;
+    }
+
+    for (const auto& pTrack : trackPointers) {
+        pTrack->setReplayGain(mixxx::ReplayGain());
+    }
+}
+
+void WTrackMenu::slotClearWaveform() {
+    const TrackPointerList trackPointers = getTrackPointers();
+    if (trackPointers.empty()) {
+        return;
+    }
+
+    AnalysisDao& analysisDao = m_pTrackCollectionManager->internalCollection()->getAnalysisDAO();
+    for (const auto& pTrack : trackPointers) {
+        analysisDao.deleteAnalysesForTrack(pTrack->getId());
+        pTrack->setWaveform(WaveformPointer());
+        pTrack->setWaveformSummary(WaveformPointer());
+    }
+}
+
+void WTrackMenu::slotClearLoop() {
+    const TrackPointerList trackPointers = getTrackPointers();
+    if (trackPointers.empty()) {
+        return;
+    }
+
+    for (const auto& pTrack : trackPointers) {
+        pTrack->removeCuesOfType(mixxx::CueType::Loop);
+    }
+}
+
+void WTrackMenu::slotClearHotCues() {
+    const TrackPointerList trackPointers = getTrackPointers();
+    if (trackPointers.empty()) {
+        return;
+    }
+
+    for (const auto& pTrack : trackPointers) {
+        pTrack->removeCuesOfType(mixxx::CueType::HotCue);
+    }
+}
+
+void WTrackMenu::slotClearAllMetadata() {
+    slotClearBeats();
+    slotClearPlayCount();
+    slotClearMainCue();
+    slotClearHotCues();
+    slotClearIntroCue();
+    slotClearOutroCue();
+    slotClearLoop();
+    slotClearKey();
+    slotClearReplayGain();
+    slotClearWaveform();
+}
+
+void WTrackMenu::slotTrackInfoClosed() {
+    DlgTrackInfo* pTrackInfo = m_pTrackInfo.take();
+    // We are in a slot directly invoked from DlgTrackInfo. Delete it
+    // later.
+    if (pTrackInfo != nullptr) {
+        pTrackInfo->deleteLater();
+    }
+}
+
+void WTrackMenu::slotTagFetcherClosed() {
+    DlgTagFetcher* pTagFetcher = m_pTagFetcher.take();
+    // We are in a slot directly invoked from DlgTagFetcher. Delete it
+    // later.
+    if (pTagFetcher != nullptr) {
+        pTagFetcher->deleteLater();
+    }
+}
+
+void WTrackMenu::slotShowTrackInfo() {
+    auto trackModel = getTrackModel();
+    if (trackModel) {
+        const auto indices = getTrackIndices();
+        showTrackInfo(indices.at(0));
+    } else {
+        const auto trackPointers = getTrackPointers();
+        showTrackInfo(trackPointers.at(0));
+    }
+}
+
+void WTrackMenu::slotNextTrackInfo() {
+    auto trackModel = getTrackModel();
+    VERIFY_OR_DEBUG_ASSERT(trackModel) {
+        return;
+    }
+    QModelIndex nextRow = currentTrackInfoIndex.sibling(
+            currentTrackInfoIndex.row() + 1, currentTrackInfoIndex.column());
+    if (nextRow.isValid()) {
+        showTrackInfo(nextRow);
+        if (!m_pTagFetcher.isNull()) {
+            showDlgTagFetcher(nextRow);
+        }
+    }
+}
```

Comment:
Thanks for the advice. This would make it much more flexible and clean. `TrackModel*` and `QModelIndex` will be made optional parameters in both of these dialogs to dynamically show or hide next/prev buttons and switch tracks internally.